core/validation/specific_validation/pathfinders/round_1.py

```diff
@@ -73,29 +74,38 @@ def _check_standard_outputs_outcomes(
     extracted_table_dfs: dict[str, pd.DataFrame], control_mappings: dict[str, dict | list[str]]
 ) -> list[Message]:
     """
-    Check that the standard outputs and outcomes in the Outputs and Outcomes tables are in the allowed values.
+    Check that the standard outputs and outcomes in the "Outputs" and "Outcomes" tables correspond to the allowed values
+    for the intervention theme selected.
 
     :param extracted_table_dfs: Dictionary of DataFrames representing tables extracted from the original Excel file
     :param control_mappings: Dictionary of control mappings extracted from the original Excel file. These mappings are
     used to validate the data in the DataFrames
     :return: List of error messages
     """
-    allowed_values = (control_mappings["standard_outputs"], control_mappings["standard_outcomes"])
+    allowed_value_dicts = (
+        control_mappings["intervention_theme_to_standard_outputs"],
+        control_mappings["intervention_theme_to_standard_outcomes"],
+    )
     worksheets = ["Outputs", "Outcomes"]
     table_names = ["Outputs", "Outcomes"]
     columns = ("Output", "Outcome")
     error_messages = []
-    for table_name, worksheet, column, allowed_values in zip(table_names, worksheets, columns, allowed_values):
+    for table_name, worksheet, column, allowed_value_dict in zip(table_names, worksheets, columns, allowed_value_dicts):
```

Comment:
This function getting pretty big and hard to read (16 locally declared variables now), and I'm not a massive fan of iterating through a zipped structure because:

- It's hard to read initially, as we are effectively "vertically slicing" a load of data structures
- The whole function is not very re-useable for future use - it's super specific and does lots of stuff
- It feels like an overly contrived method to loop through 2 rows.

Not sure how better to do it though? Maybe change the input data into an overly verbose dict of values, or abstract out the bit after the initial loop into a separate function that takes all the params, then just verbosely call it twice with the params explicitly stated as kwargs for each call? Or even a combination of the 2? None of these feel great though... It's always a dilemma in Python when trying to write such dynamic functions.

Reply:
just my thoughts on this one, no need to action if you can't easily think of anything